src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/Platform/SqlCreateDatabaseWriter.cs

```diff
@@ -15,6 +16,15 @@ public SqlCreateDatabaseWriter(string databaseName)
         public string DatabaseName => _databaseName;
 
         public string GetSql()
-            => Invariant($"CREATE DATABASE [{_databaseName}]");
+        {
+            var sql = new StringBuilder();
+
+            sql.AppendLine(Invariant($"IF NOT EXISTS (SELECT name FROM sys.databases WHERE name = '{_databaseName}')"));
+            sql.AppendLine("BEGIN");
+            sql.AppendLine(Invariant($"CREATE DATABASE [{_databaseName}]"));
+            sql.AppendLine("END");
```

Comment:
Looking at the permissions for `sys.databases` everything should be good, unless someone has changed the default database for their login to something other than master and granted `CREATE DATABASE` but not `ALTER ANY DATABASE` or `VIEW ANY DATABASE`. Sounds very unlikely, since auto creation should really only be used in a dev/test environment.

> If the caller of sys.databases isn't the owner of the database and the database isn't master or tempdb, the minimum permissions required to see the corresponding row are ALTER ANY DATABASE or the VIEW ANY DATABASE server-level permission, or CREATE DATABASE permission in the master database. The database to which the caller is connected can always be viewed in sys.databases.
> (https://learn.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-databases-transact-sql?view=sql-server-ver16)